src/java.base/share/classes/sun/security/tools/keytool/Main.java

```diff
@@ -405,26 +405,39 @@ public String toString() {
         collator.setStrength(Collator.PRIMARY);
     }
 
-    private Main() { }
-
     public static void main(String[] args) throws Exception {
         Main kt = new Main();
-        kt.run(args, System.out);
+        try {
+            kt.run(args, System.out);
+        } catch (ExitException ee) {
+            System.exit(ee.errorCode);
+        }
+    }
+
+    private static class ExitException extends RuntimeException {
+        @java.io.Serial
+        static final long serialVersionUID = 0L;
+        private final int errorCode;
+        public ExitException(int errorCode) {
+            this.errorCode = errorCode;
+        }
     }
 
-    private void run(String[] args, PrintStream out) throws Exception {
+    public void run(String[] args, PrintStream out) throws Exception {
         try {
-            args = parseArgs(args);
+            parseArgs(args);
             if (command != null) {
                 doCommands(out);
             }
+        } catch (ExitException ee) {
+            throw ee;
```

Comment:
`ExitException` is a private inner class, so external users won’t be able to distinguish it from a regular `Exception` to extract the `errorCode`.